code/modules/research/xenobiology/crossbreeding/warping.dm

```diff
@@ -0,0 +1,918 @@
+	..()
+	if(!istype(AM,/mob/living/carbon/human))
+		return
+	H = AM
+	H.dna.species.punchstunthreshold += 20//we don't want them to insta stun everyone.
+	H.dna.species.punchdamagelow += 20
+	H.dna.species.punchdamagehigh += 20 //buffed up punch damage
+	to_chat(H, "<span class='warning'>You feel the urge to punch something really hard!</span>")
+
+
+///takes away the punch damage when you leave
+/obj/effect/warped_rune/redspace/Uncrossed(atom/movable/AM)
+	if(!istype(AM,/mob/living/carbon/human)) //checks if the person that just left is the same as the currently enraged person
+		return
+	H = AM
+	H.dna.species.punchstunthreshold -= 20
+	H.dna.species.punchdamagelow -= 20
+	H.dna.species.punchdamagehigh -= 20
+
+
+///destroying the rune will also remove the punch force of the persons on the rune.
+/obj/effect/warped_rune/redspace/Destroy()
+	..()
+	for(var/mob/living/carbon/human/D in T) // takes away people's punch damage that are on the rune anyway.
+		H.dna.species.punchstunthreshold -= 20
+		H.dna.species.punchdamagelow -= 20
+		H.dna.species.punchdamagehigh -= 20.
+
+
+
+
+
+/* Green rune vores plasma and spews out xeno resin which lets you build xeno structure*/
+
+/obj/item/slimecross/warping/green
+	colour = "green"
+	runepath = /obj/effect/warped_rune/greenspace
+	drawing_time = 100
+
+/obj/effect/warped_rune/greenspace
+	icon_state = "xeno_rune"
+	desc = "We will build walls out of our fallen foes, they shall fear our very buildings."
+	var/obj/item/stack/sheet/xeno_resin/X
+	max_cooldown = 100
+
+
+
+/obj/item/stack/sheet/xeno_resin
+	name = "Resin sheets"
+	icon = 'icons/mob/alien.dmi'
+	icon_state = "nestoverlay" //literally just the xeno nest icon
+	merge_type = /obj/item/stack/sheet/xeno_resin
+
+
+
+/obj/effect/warped_rune/greenspace/Initialize()
+	.=..()
+	START_PROCESSING(SSprocessing, src)
+
+
+/obj/effect/warped_rune/greenspace/process()
+	if(cooldown < world.time)
+		cooldown = world.time + max_cooldown
+		transmute_resin()
+
+/obj/effect/warped_rune/greenspace/proc/transmute_resin()
+	for(var/obj/item/stack/sheet/mineral/plasma/P in T)
+		X = new(T)
+		P.amount--
+		if(P.amount <= 0)
+			qdel(P)
+
+//note : some of these can only be built ON resin weeds such as the resin nest.
+GLOBAL_LIST_INIT(resin_recipes, list ( \
+	new/datum/stack_recipe("Resin seed", /obj/structure/alien/weeds/node, 10, one_per_turf = 1, on_floor = 1, time = 100), \
+	new/datum/stack_recipe("Resin Wall", /obj/structure/alien/resin/wall, 3, one_per_turf = 1, on_floor = 1 , time = 50), \
+	new/datum/stack_recipe("Fake hatched egg", /obj/structure/alien/egg/burst, 2, one_per_turf = 1, on_floor = 1), \
+	new/datum/stack_recipe("Resin nest", /obj/structure/bed/nest , 5, one_per_turf = 1, on_floor = 0, time = 50), \
+	))
+
+/obj/item/stack/sheet/xeno_resin/get_main_recipes()
+	.= ..()
+	. += GLOB.resin_recipes
+
+
+
+
+
+
+
+/* pink rune, makes people slightly happier after walking on it*/
+
+/obj/item/slimecross/warping/pink
```

Comment:
The formatting on this PR is quite an eye sore. Try improving it some, use some kind of format and stick to it. Be consistent.
If you have none try using spaces between the operators (`. = ..()` instead of mixed cases `.= ..()` or `.=..()`), and put one or two spaces between each object or proc definition instead of a random amount.